nisystemlink/clients/core/_uplink/_base_client.py

```diff
@@ -1,5 +1,29 @@
+# mypy: disable-error-code = misc
+
+from typing import Callable, cast, NoReturn, Optional
+
 from nisystemlink.clients import core
-from uplink import Consumer
+from requests import JSONDecodeError, Response
+from uplink import Consumer, response_handler
+
+
+@response_handler
+def _handle_http_status(response: Response) -> Optional[Response]:
+    """Checks an HTTP response's status code and raises an exception if necessary."""
+    if 200 <= response.status_code < 300:
+        # Return None for "204 No Content" responses.
+        if response.status_code == 204:
+            return None
+        return response
+    try:
+        content = response.json()
+        if isinstance(content, dict) and "error" in content:
+            error = core.ApiError.from_json_dict(content["error"])
+            raise core.ApiException(error=error, http_status_code=response.status_code)
+        else:
+            cast(Callable[[], NoReturn], response.raise_for_status)()
```

Comment:
mypy was complaining that this method was missing returns because `raise_for_status` wasn't typed as `NoReturn`, so I had to add the cast

Reply:
I refactored this to be more like how the tag API handles things.